lib/nice_partials/partial.rb

```diff
@@ -49,8 +55,44 @@ def capture(block)
 
     private
 
+    class Section
+      def initialize(view_context)
+        @view_context = view_context
+        @pending_content = nil
+        @content = ActiveSupport::SafeBuffer.new
+      end
+
+      def content_for(*arguments, &block)
+        if write_content_for(arguments.first, &block)
+          nil
+        else
+          capture_content_for(*arguments) if @pending_content
+          @content.presence
+        end
+      end
+
+      def content?
+        @pending_content.present? || @content.present?
+      end
+
+      private
+
+      def write_content_for(content = nil, &block)
+        if content && !@pending_content
+          @content << content.to_s
+        else
+          @pending_content = block if block
+        end
+      end
+
+      def capture_content_for(*arguments)
+        @content << @view_context.capture(*arguments, &@pending_content).to_s
+        @pending_content = nil
+      end
+    end
+
     def contents
-      @contents ||= Hash.new { |h, k| h[k] = ActiveSupport::SafeBuffer.new }
+      @contents ||= Hash.new { |h, k| h[k] = Section.new(@view_context) }
```

Comment:
Just for reference, the idea was to spare the Hash allocation so we could keep a `Partial` allocation leaner, since we'll be probably be doing one of those per `render` call. But in the grand scheme of things it's probably negligible.